receiver/otlpreceiver/internal/arrow/arrow.go

```diff
@@ -135,64 +146,85 @@ func newHeaderReceiver(streamCtx context.Context, includeMetadata bool) *headerR
 // client.Info with additional key:values associated with the arrow batch.
 // This is safe to call when h is nil.
 func (h *headerReceiver) combineHeaders(ctx context.Context, hdrsBytes []byte) (context.Context, map[string][]string, error) {
-	if h == nil || (len(hdrsBytes) == 0 && len(h.streamHdrs) == 0) {
+	if len(hdrsBytes) == 0 && len(h.streamHdrs) == 0 {
 		return ctx, nil, nil
 	}
 
 	if len(hdrsBytes) == 0 {
 		return h.newContext(ctx, h.streamHdrs), h.streamHdrs, nil
 	}
 
-	h.tmpHdrs = map[string][]string{}
+	// Note that we will parse the headers even if they are not
+	// used, to check for validity.  tmpHdrsAppend() will skip
+	// modifying tmpHdrs if it is nil.
+	h.tmpHdrs = nil
+
+	if h.includeMetadata || h.hasAuthServer {
+		h.tmpHdrs = map[string][]string{}
+	}
 
 	// Write calls the emitFunc, appending directly into `tmpHdrs`.
 	if _, err := h.decoder.Write(hdrsBytes); err != nil {
 		return ctx, nil, err
 	}
 
-	// Add streamHdrs that were not carried in the per-request headers.
-	for k, v := range h.streamHdrs {
-		// Note: This is done after the per-request metadata is defined
-		// in recognition of a potential for duplicated values stemming
-		// from the Arrow exporter's independent call to the Auth
-		// extension's GetRequestMetadata().  This paired with the
-		// headersetter's return of empty-string values means, we would
-		// end up with an empty-string element for any headersetter
-		// `from_context` rules b/c the stream uses background context.
-		// This allows static headers through.
-		//
-		// See https://github.com/open-telemetry/opentelemetry-collector/issues/6965
-		if _, ok := h.tmpHdrs[k]; !ok {
-			h.tmpHdrs[k] = v
+	if h.tmpHdrs != nil {
```

Comment:
given that checking the existence of a key in a nil map will [always return false](https://go.dev/play/p/sk-vjkwRx6e), are we doing this nil check to short circuit so we don't iterate over the stream headers unnecessarily?